tests/integ/modin/series/test_cache_result.py

```diff
@@ -0,0 +1,168 @@
+    """
+    series = series.reset_index(drop=True)
+    return module.concat([series] * 10)
+
+
+@pytest.fixture(scope="function")
+def simple_test_data():
+    int_data = list(range(9))
+    str_data = ["foo", "bar", "baz", "foo", "bar", "baz", "foo", "bar", "baz"]
+    return list(chain.from_iterable(zip(int_data, str_data)))
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+def test_cache_result_empty_series(inplace):
+    native_series = native_pd.Series()
+    snow_series = pd.Series()
+    snow_series_copy = snow_series.copy(deep=True)
+    with SqlCounter(query_count=1):
+        cached_snow_series = cache_and_return_series(snow_series, inplace)
+    with SqlCounter(query_count=2):
+        assert_series_equal(
+            cached_snow_series.to_pandas(), snow_series_copy.to_pandas()
+        )
+    with SqlCounter(query_count=1):
+        assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+            snow_series, native_series
+        )
+
+    native_series = native_pd.Series(index=["A", "B", "C"])
+    snow_series = pd.Series(native_series)
+    snow_series_copy = snow_series.copy(deep=True)
+    with SqlCounter(query_count=1):
+        cached_snow_series = cache_and_return_series(snow_series, inplace)
+    with SqlCounter(query_count=2):
+        assert_series_equal(
+            cached_snow_series.to_pandas(), snow_series_copy.to_pandas()
+        )
+    with SqlCounter(query_count=1):
+        assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+            cached_snow_series, native_series
+        )
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+def test_cache_result_series_complex_correctness(time_index_series_data, inplace):
+    series_data, kwargs = time_index_series_data
+    snow_series, native_series = create_test_series(series_data, **kwargs)
+    snow_series = snow_series.resample("2H").mean()
+    snow_series_copy = snow_series.copy(deep=True)
+    with SqlCounter(query_count=1):
+        cached_snow_series = cache_and_return_series(snow_series, inplace)
+    with SqlCounter(query_count=2):
+        assert_series_equal(
+            cached_snow_series.to_pandas(), snow_series_copy.to_pandas()
+        )
+    native_series = native_series.resample("2H").mean()
+
+    with SqlCounter(query_count=1):
+        cached_snow_series = cached_snow_series.diff()
+        native_series = native_series.diff()
+        assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(
+            cached_snow_series, native_series, check_freq=False
+        )
+
+
+@pytest.mark.parametrize("inplace", [True, False])
+class TestCacheResultReducesQueryCount:
+    def test_cache_result_simple(self, inplace):
+        snow_series = pd.concat(
+            [pd.Series(list(range(i, i + 5))) for i in range(0, 50, 5)]
+        )
+        native_series = perform_chained_operations(
+            native_pd.Series(np.arange(50)), native_pd
+        )
+        native_series.name = 0
```

Comment:
why is this line here? We don't do the same to the Snowpark pandas series. Is it working around a Snowpark pandas bug? if so, can you link the bug here?

Reply:
Seems like this is a bug in snowpark pandas - filing a JIRA and linking it here.